# What drives underpricing in Indian IPOs?

**Dataset:** 509 Indian IPOs from January 2010 to August 2025, with 19 columns: offer details, subscription levels, market conditions and the underpricing result. 
Source: [https://www.kaggle.com/datasets/ishans474/indian-ipo-underpricing-2010-2025].

**Key columns:**
- Issue_Size(crores): size of the issue
- QIB, HNI, RII, Total: subscription multiples for institutional investors, high-net-worth investors, retail investors and overall
- NIFTY_20D_Return, NIFTY_60D_Return: Nifty index returns over 20 and 60 days after the IPO date (market conditions, not the IPO's own return; verify in the dataset description)
- India_VIX: market fear index at the time of the IPO
- Underpricing: percentage gain at listing compared to the offer price (verify in the dataset description)
- Underpricing_Class: Low / Medium / High category of underpricing

**Main question:** Are heavily subscribed IPOs more underpriced, and do market conditions (VIX) make a difference?

**Tools:** Python, pandas.

In [115]:
#importing pandas
import pandas as pd
pd.set_option('display.max_columns', None)


In [116]:
df = pd.read_csv("IPO_Underpricing_Final.csv")

In [117]:
print("total number of rows and columns in the dataset:", df.shape)

total number of rows and columns in the dataset: (509, 19)


In [118]:
print("total number of IPOs in the dataset:",df['IPO_Name'].nunique())
#There are 508 unique IPO names across 509 rows, so one name appears twice.

total number of IPOs in the dataset: 508


In [119]:
print("total number of duplicate names in dataset:",df['IPO_Name'].duplicated().sum())

total number of duplicate names in dataset: 1


In [120]:
print("name of duplicate IPOs in the dataset:",df[df['IPO_Name'].duplicated()]['IPO_Name'].unique())

name of duplicate IPOs in the dataset: <StringArray>
['Power Grid Corporation of India Ltd']
Length: 1, dtype: str


In [121]:
print("total number of null issue size is :",df['Issue_Size(crores)'].isnull().sum())
#There are no missing values in Issue_Size(crores).


total number of null issue size is : 0


In [122]:
highest_row = df.loc[df['Issue_Size(crores)'].idxmax()]
print(highest_row)
# The largest issue was Life Insurance Corporation of India (LIC) in May 2022: Rs 20,557 crore at an offer price of Rs 949.
#  It was only subscribed 1.64 times overall and listed below its offer price (underpricing of about -8.6%). The market was also nervous (India VIX 24.5). 
# So the biggest IPO in the data was one of the weakest performers.

Date                                                     2022-05-17
IPO_Name                  Life Insurance Corporation of India (LIC)
Issue_Size(crores)                                          20557.2
Offer Price                                                   949.0
QIB                                                            2.83
HNI                                                            1.39
RII                                                            1.14
Total                                                          1.64
NIFTY_20D_Return                                          -9.346437
NIFTY_60D_Return                                          -5.940229
NIFTY_20D_Volatility                                       0.011573
India_VIX                                                 24.530001
Log_Issue_Size(crores)                                     9.931015
Log_QIB                                                    1.342865
Log_HNI                                         

In [ ]:
# IPO launched before the highest 60-day Nifty return
highest_returner= df.loc[df['NIFTY_60D_Return'].idxmax()]
print(highest_returner)

Date                                             2020-12-24
IPO_Name                  Mrs Bectors Food Specialities Ltd
Issue_Size(crores)                                   540.54
Offer Price                                           288.0
QIB                                                  176.85
HNI                                                  620.86
RII                                                   29.33
Total                                                198.02
NIFTY_20D_Return                                   4.181868
NIFTY_60D_Return                                   23.08409
NIFTY_20D_Volatility                               0.010085
India_VIX                                              20.5
Log_Issue_Size(crores)                             6.294417
Log_QIB                                             5.18094
Log_HNI                                            6.432715
Log_RII                                            3.412137
Log_Total                               

In [124]:
# to get all low underpricing ones
low = df[df['Underpricing_Class']=='Low']
print(low)
# 300 IPOs fall in the Low underpricing class. 
# Low includes negative values, meaning IPOs that listed below their offer price.

           Date                                   IPO_Name  \
0    2010-01-04                         JSW Energy Limited   
1    2010-01-05                  Godrej Properties Limited   
3    2010-01-11                 MBL Infrastructure Limited   
4    2010-01-29               Birla Shloka Edutech Limited   
5    2010-02-03        Infinite Computer Solutions Limited   
..          ...                                        ...   
501  2025-07-30                    Indiqube Spaces Limited   
503  2025-07-31             Brigade Hotel Ventures Limited   
505  2025-08-05                    Laxmi India Finance Ltd   
506  2025-08-06  National Securities Depository Ltd (NSDL)   
508  2025-08-06                      M & B Engineering Ltd   

     Issue_Size(crores)  Offer Price     QIB     HNI    RII  Total  \
0               2700.00        100.0    2.88    0.15   0.40   1.68   
1                468.85        490.0    7.45    0.41   0.38   4.00   
3                102.60        180.0    3.35 

In [125]:
# to find top 5 largest ipos
largest_top5 = df.sort_values(by=['Issue_Size(crores)'], ascending=False)
print(largest_top5.head(5))
#The five largest issues were LIC (Rs 20,557 cr), Vodafone Idea (Rs 18,000 cr), Coal India (Rs 15,199 cr), Yes Bank (Rs 15,000 cr) and HDB Financial Services (Rs 12,500 cr)

           Date                                   IPO_Name  \
313  2022-05-17  Life Insurance Corporation of India (LIC)   
420  2024-04-25                         Vodafone Idea Ltd    
62   2010-11-04                             Coal India Ltd   
246  2020-07-27                               Yes Bank Ltd   
496  2025-07-02             HDB Financial Services Limited   

     Issue_Size(crores)  Offer Price    QIB    HNI   RII  Total  \
313             20557.2        949.0   2.83   1.39  1.14   1.64   
420             18000.0         11.0  17.56   4.13  0.92   6.36   
62              15199.4        245.0  24.70  25.40  2.31  15.28   
246             15000.0         12.0   1.90   0.63  0.47   0.93   
496             12500.0        740.0  55.47   9.99  1.41  16.69   

     NIFTY_20D_Return  NIFTY_60D_Return  NIFTY_20D_Volatility  India_VIX  \
313         -9.346437         -5.940229              0.011573  24.530001   
420          1.773815          5.478650              0.005985  10.280000

In [126]:
# IPOs launched when the Nifty rose over both the next 20 and 60 days
bothprofit = df[(df['NIFTY_60D_Return']>0) & (df['NIFTY_20D_Return']>0)];
print(bothprofit)

# 247 IPOs were launched when the Nifty rose over both the next 20 and 60 days.
#  This describes market conditions after the IPO date, not the profits made by the IPOs themselves.

           Date                          IPO_Name  Issue_Size(crores)  \
0    2010-01-04                JSW Energy Limited             2700.00   
1    2010-01-05         Godrej Properties Limited              468.85   
2    2010-01-06                  D B Corp Limited              384.22   
3    2010-01-11        MBL Infrastructure Limited              102.60   
17   2010-03-10  Texmo Pipes and Products Limited               45.00   
..          ...                               ...                 ...   
495  2025-07-02        Sambhv Steel Tubes Limited              540.00   
496  2025-07-02    HDB Financial Services Limited            12500.00   
497  2025-07-03     Indogulf Cropsciences Limited              200.00   
498  2025-07-09                    Crizac Limited              860.00   
499  2025-07-17      Smartworks Coworking Limited              582.56   

     Offer Price     QIB    HNI    RII  Total  NIFTY_20D_Return  \
0          100.0    2.88   0.15   0.40   1.68          1

In [127]:
#to know how many are launched everyyear
df['Date'] = pd.to_datetime(df['Date'])
ipo_counts = df['Date'].dt.year.value_counts().sort_index()
print(ipo_counts)

#IPO activity varied a lot: 72 in 2010, a low of 4 in 2013, and a high of 75 in 2024. Activity dropped sharply from 2012 to 2014 and recovered from 2021 onwards. 
# 2025 is a partial year (data ends in August).

Date
2010    72
2011    39
2012    10
2013     4
2014     6
2015    18
2016    26
2017    34
2018    19
2019    16
2020    13
2021    47
2022    39
2023    54
2024    75
2025    37
Name: count, dtype: int64


In [128]:
# to know how many are energies IPOs
energy_ipos = df[df['IPO_Name'].str.contains('Energ', case=False, na=False)]
print(energy_ipos[['IPO_Name', 'Issue_Size(crores)']])

# 9 IPOs have "Energ" in their name, such as JSW Energy and NTPC Green Energy.
#  This is only a rough proxy because the dataset has no sector column: it misses energy companies without the word in their name and includes companies like Ather Energy (electric scooters).

                                              IPO_Name  Issue_Size(crores)
0                                   JSW Energy Limited             2700.00
108                                   Ujaas Energy Ltd               93.00
274                  Shyam Metalics and Energy Limited              908.55
334                          Inox Green Energy Limited              740.00
377                                 IRM Energy Limited              544.36
384  Indian Renewable Energy Development Agency Lim...             2150.21
433                           Premier Energies Limited             2830.40
457                          NTPC Green Energy Limited            10000.00
483                               Ather Energy Limited             2980.80


In [129]:
df["Date"] = pd.to_datetime(df["Date"])
df["Year"] = df["Date"].dt.year

In [130]:
# to find Average underpricing by year (groupby) .
df.groupby("Year")["Underpricing"].agg(["count", "mean", "median"]).round(2)

#Median underpricing by year: 
#- Highest in 2020 (49%), but only 13 IPOs, so this is a small sample.
#- Negative in 2018 (-0.14%) and exactly 0% in 2015, meaning many IPOs listed flat or lower.
#- In several years the mean is much higher than the median (2017: 20.7% vs 3.1%; 2021: 24.5% vs 5.7%). A few very large listings pull the mean up, so the median describes the typical IPO better.
#- 2013 (4 IPOs) and 2014 (6 IPOs) are too small to draw conclusions from.

,count,mean,median
Year,,,
2010,72,10.08,6.62
2011,39,3.18,1.35
2012,10,2.41,0.91
2013,4,4.50,5.37
2014,6,25.13,18.84
2015,18,5.78,0.00
2016,26,11.48,8.84
2017,34,20.68,3.13
2018,19,1.32,-0.14


In [131]:
df[["Underpricing", "Total", "QIB", "India_VIX", "Issue_Size(crores)"]].corr()

#Underpricing is strongly positively correlated with overall subscription (0.72) and QIB subscription (0.60).
#  It has a weak negative link with India VIX (-0.17) and issue size (-0.10). Total and QIB are themselves strongly correlated (0.81), because institutional demand is part of total demand, so they should not be treated as independent evidence. 

,Underpricing,Total,QIB,India_VIX,Issue_Size(crores)
Underpricing,1.000000,0.715313,0.596749,-0.173953,-0.097491
Total,0.715313,1.000000,0.808560,-0.209112,-0.147012
QIB,0.596749,0.808560,1.000000,-0.282313,-0.077523
India_VIX,-0.173953,-0.209112,-0.282313,1.000000,-0.005807
Issue_Size(crores),-0.097491,-0.147012,-0.077523,-0.005807,1.000000


In [132]:
df["Sub_Group"] = pd.qcut(df["Total"], q=3, labels=["Low", "Medium", "High"])

In [133]:
df.groupby("Sub_Group", observed=True)["Total"].agg(["count", "min", "max"])

#IPOs were split into three equal groups by total subscription: Low (up to 2.97 times), Medium (2.98 to 30.57 times) and High (30.63 to 326 times).

,count,min,max
Sub_Group,,,
Low,170,0.11,2.97
Medium,169,2.98,30.57
High,170,30.63,326.49


In [134]:
# to find if high-subscription IPOs have higher underpricing?
df.groupby("Sub_Group", observed=True)["Underpricing"].agg(["count", "mean", "median"]).round(2)

#Underpricing rises steadily with subscription. Median underpricing is 0% in the Low group, 4.3% in the Medium group and 31.2% in the High group (means: -0.8%, 7.4%, 41.5%).
#  Heavily subscribed IPOs are, on average, listed at much higher prices than their offer price. The High group covers a wide range (30x to 326x), so there is variation inside it.

,count,mean,median
Sub_Group,,,
Low,170,-0.83,0.00
Medium,169,7.39,4.29
High,170,41.45,31.21


In [135]:
# to find if IPOs in high-VIX periods show lower underpricing
df["VIX_Group"] = pd.qcut(df["India_VIX"], q=3, labels=["Low", "Medium", "High"])
df.groupby("VIX_Group", observed=True)["Underpricing"].agg(["count", "mean", "median"]).round(2)

#IPOs were split into three equal groups by India VIX. Median underpricing falls as market fear rises: 10.6% in Low-VIX periods, 6.0% in Medium and 2.9% in High.
#  IPOs in calmer markets tend to be more underpriced. The effect is smaller than the subscription effect.

,count,mean,median
VIX_Group,,,
Low,170,20.78,10.59
Medium,169,16.72,6.02
High,170,10.56,2.90


## Conclusions

1. **Subscription level is the strongest signal.** Median underpricing is 0% for IPOs subscribed under 3 times, versus 31% for those subscribed over 30 times.
2. **Market conditions matter, but less.** Median underpricing falls from 10.6% in low-VIX periods to 2.9% in high-VIX periods.
3. **Issue size shows little link** to underpricing (correlation -0.10). The largest IPO, LIC, listed below its offer price.
4. **Average figures are misleading.** A few extreme listings push the mean well above the median in many years, so medians are used throughout.

## Limitations
- These are associations, not proof of cause. Strong demand and high listing gains may both reflect the same market enthusiasm.
- Some years have very few IPOs (2013: 4, 2014: 6), and 2025 ends in August.
- Subscription and VIX are related to each other, which makes their effects hard to fully separate.
- 509 IPOs from a single dataset is a modest sample, and the energy filter is based only on company names.